src/runmode-dpdk.c

```diff
@@ -0,0 +1,1274 @@
+                rxq_conf.rx_thresh.hthresh, rxq_conf.rx_thresh.pthresh, rxq_conf.rx_thresh.wthresh,
+                rxq_conf.rx_free_thresh, rxq_conf.rx_drop_en, rxq_conf.offloads);
+
+        retval = rte_eth_rx_queue_setup(iconf->port_id, queue_id, iconf->nb_rx_desc,
+                iconf->socket_id, &rxq_conf, iconf->pkt_mempool_array[queue_id]);
+        if (retval < 0) {
+            SCLogError(SC_ERR_DPDK_INIT,
+                    "Error (err=%d) during initialization of device queue %u of port %u", retval,
+                    queue_id, iconf->port_id);
+            SCReturnInt(retval);
+        }
+    }
+
+    for (uint16_t queue_id = 0; queue_id < iconf->nb_rx_queues; queue_id++) {
+        txq_conf = dev_info->default_txconf;
+        txq_conf.offloads = port_conf->txmode.offloads;
+        retval = rte_eth_tx_queue_setup(
+                iconf->port_id, queue_id, iconf->nb_tx_desc, iconf->socket_id, &txq_conf);
+        if (retval < 0) {
+            SCLogError(SC_ERR_DPDK_INIT,
+                    "Error (err=%d) during initialization of device queue %u of port %u", retval,
+                    queue_id, iconf->port_id);
+            SCReturnInt(retval);
+        }
+    }
+
+    SCReturnInt(0);
+}
+
+int DeviceConfigureIPS(DPDKIfaceConfig *iconf)
+{
+    int retval;
+
+    if (iconf->out_iface != NULL) {
+        retval = rte_eth_dev_get_port_by_name(iconf->out_iface, &iconf->out_port_id);
+        if (retval != 0) {
+            SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) during obtaining port id of %s", retval,
+                    iconf->out_iface);
+            SCReturnInt(retval);
+        }
+
+        if (rte_eth_dev_socket_id(iconf->port_id) != rte_eth_dev_socket_id(iconf->out_port_id)) {
+            SCLogWarning(SC_WARN_DPDK_CONF, "%s and %s are not on the same NUMA node", iconf->iface,
+                    iconf->out_iface);
+        }
+    }
+    SCReturnInt(0);
+}
+
+int DeviceConfigure(DPDKIfaceConfig *iconf)
+{
+    // configure device
+    int retval;
+    struct rte_eth_dev_info dev_info;
+    struct rte_eth_conf port_conf;
+
+    retval = rte_eth_dev_get_port_by_name(iconf->iface, &(iconf->port_id));
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) when getting port id of %s Is device enabled?",
+                retval, iconf->iface);
+        SCReturnInt(retval);
+    }
+
+    if (!rte_eth_dev_is_valid_port(iconf->port_id)) {
+        SCLogError(SC_ERR_DPDK_INIT, "Specified port %d is invalid", iconf->port_id);
+        SCReturnInt(retval);
+    }
+
+    retval = rte_eth_dev_socket_id(iconf->port_id);
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_INIT, "Error (err=%d) invalid socket id (port %u)", retval,
+                iconf->port_id);
+        SCReturnInt(retval);
+    } else {
+        iconf->socket_id = retval;
+    }
+
+    retval = rte_eth_dev_info_get(iconf->port_id, &dev_info);
+    if (retval != 0) {
```

Comment:
as mentioned in yaml and parse comments, please re-work to ensure all validations are done before configuration is done. If there is a criteria mismatch return failure gracefully. Re-work